### PR TITLE
update charts/yaml/release notes to 3.1.6

### PR DESCRIPTION
**Pull request by wanlonghenry**
containerinsights agent 3.1.6 release, update charts/yaml/release notes to 3.1.6

### CODE REVIEW

File: ReleaseNotes.md [hunk 1]
```diff
@@ -8,6 +8,32 @@ information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeo
 additional questions or comments.
 
 ## Release History
+### 04/07/2023 -
+##### Version microsoft/oms:3.1.6 Version mcr.microsoft.com/azuremonitor/containerinsights/ciprod:3.1.6 (linux)
+##### Version microsoft/oms:win-3.1.6 Version mcr.microsoft.com/azuremonitor/containerinsights/ciprod:win-3.1.6 (windows)
+##### Code change log
+- Common
+  - Update prometheus.io/scheme default description in configmap
```
Comment (pfrcks) [inline ReleaseNotes.md:16]: schema

File: ReleaseNotes.md [hunk 2]
```diff
@@ -8,6 +8,32 @@ information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeo
...
+  - Add support to specify resource endpoint for metrics 
+  - Add autonomous fqdn to endpoint for metrics 
+  - Add devskim as github action and codeql as enabled in ADO pipeline
+  - AgentLogCollection.sh update for collecting more detailed logs
+  - Fix pod ready condition issue for pods that are job ready 
+  - Updating packages for go vulnerabilities
+  - Remove the necessary askcoin references in troubleshooter and readme
+  - Change md5 to sha256
```
Comment (pfrcks) [inline ReleaseNotes.md:24]: *to omslog

File: ReleaseNotes.md [hunk 3]
```diff
@@ -8,6 +8,32 @@ information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeo
 additional questions or comments.
 
 ## Release History
+### 04/07/2023 -
+##### Version microsoft/oms:3.1.6 Version mcr.microsoft.com/azuremonitor/containerinsights/ciprod:3.1.6 (linux)
+##### Version microsoft/oms:win-3.1.6 Version mcr.microsoft.com/azuremonitor/containerinsights/ciprod:win-3.1.6 (windows)
+##### Code change log
+- Common
+  - Update prometheus.io/scheme default description in configmap
+  - Add support to specify resource endpoint for metrics
```
Comment (ganga1980) [inline ReleaseNotes.md:17]: This change also specific to autonomous.  So update this to "Add support to specify  azure autonomous resource endpoint parameter for the Extension chart for custom metrics "

File: ReleaseNotes.md (hunk 1 above)
Comment (wanlonghenry) [reply ReleaseNotes.md:16]: the wording should be right since it is coming from configmap

File: ReleaseNotes.md (hunk 2 above)
Comment (ganga1980) [inline ReleaseNotes.md:24]: update to "change md5 to sha256 for the omslogger"

Comment (wanlonghenry) [reply ReleaseNotes.md:24]: changed

Comment (wanlonghenry) [reply ReleaseNotes.md:24]: changed

File: ReleaseNotes.md (hunk 3 above)
Comment (wanlonghenry) [reply ReleaseNotes.md:17]: changed

File: ReleaseNotes.md [hunk 4]
```diff
@@ -8,6 +8,32 @@ information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeo
...
+  - Geneva logs integration for single and multi tenancy
+  - Add gcs region support in config
+  - Change output fwd plugin settings configurable
+  - Add multiline logs feature with configmap
+  - Config AMA specific envvars only if 1p mode is configured
+  - Use -p when creating dirs in main.sh
+  - Disable telegraf on replicaset by default
+  - Remove unused default gem openssl & python
+- Windows Agent
+  - Add nil check in kubelet
```
Comment (ganga1980) [inline ReleaseNotes.md:36]: This Fix applicable for both Linux & windows.

File: ReleaseNotes.md [hunk 5]
```diff
@@ -8,6 +8,32 @@ information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeo
...
+  - Change md5 to sha256 for the omslogger
+  - Log message and yaml to sync with AKS RP
+- Linux Agent
+  - Geneva logs integration for single and multi tenancy
+  - Add gcs region support in config
+  - Change output fwd plugin settings configurable
+  - Add multiline logs feature with configmap
+  - Config AMA specific envvars only if 1p mode is configured
+  - Use -p when creating dirs in main.sh
+  - Disable telegraf on replicaset by default
```
Comment (ganga1980) [inline ReleaseNotes.md:33]: For resource optimization, telegraf in replicaset only enabled either the Cluster level Prometheus scraping or NPM configured.

File: ReleaseNotes.md [hunk 6]
```diff
@@ -8,6 +8,32 @@ information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeo
...
+  - Add devskim as github action and codeql as enabled in ADO pipeline
+  - AgentLogCollection.sh update for collecting more detailed logs
+  - Fix pod ready condition issue for pods that are job ready 
+  - Updating packages for go vulnerabilities
+  - Remove the necessary askcoin references in troubleshooter and readme
+  - Change md5 to sha256 for the omslogger
+  - Log message and yaml to sync with AKS RP
+- Linux Agent
+  - Geneva logs integration for single and multi tenancy
+  - Add gcs region support in config
```
Comment (ganga1980) [inline ReleaseNotes.md:28]: This can be removed.

File: kubernetes/linux/Dockerfile [hunk 7]
```diff
@@ -19,7 +19,7 @@ RUN /usr/bin/apt-get update && /usr/bin/apt-get install -y libc-bin wget openssl
 COPY setup.sh main.sh defaultpromenvvariables defaultpromenvvariables-rs defaultpromenvvariables-sidecar mdsd.xml envmdsd logrotate.conf $tmpdir/
```
Comment (ganga1980) [inline kubernetes/linux/Dockerfile:19]: FYI.. we should clean up Dockerfile since this not being used.

File: ReleaseNotes.md (hunk 4 above)
Comment (wanlonghenry) [reply ReleaseNotes.md:36]: changed

File: ReleaseNotes.md (hunk 5 above)
Comment (wanlonghenry) [reply ReleaseNotes.md:33]: added

File: ReleaseNotes.md (hunk 6 above)
Comment (wanlonghenry) [reply ReleaseNotes.md:28]: removed

File: ReleaseNotes.md [hunk 8]
```diff
@@ -8,6 +8,30 @@ information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeo
...
+  - Add devskim as github action and codeql as enabled in ADO pipeline
+  - AgentLogCollection.sh update for collecting more detailed logs
+  - Fix pod ready condition issue for pods that are job ready 
+  - Updating packages for go vulnerabilities
+  - Remove the necessary askcoin references in troubleshooter and readme
+  - Change md5 to sha256 for the omslogger
+  - Log message and yaml to sync with AKS RP
+  - Add nil check in kubelet
+- Linux Agent
+  - Geneva logs integration for single and multi tenancy
```
Comment (ganga1980) [inline ReleaseNotes.md:28]: Following changes applicable for both windows and linux.
  - Geneva logs integration for single and multi tenancy
  - Change output fwd plugin settings configurable
  - Add multiline logs feature with configmap
  - Config AMA specific envvars only if 1p mode is configured

Comment (wanlonghenry) [reply ReleaseNotes.md:28]: in this case do we want to add windows tag in the geneva values.yaml?

File: kubernetes/linux/Dockerfile (hunk 7 above)
Comment (wanlonghenry) [reply kubernetes/linux/Dockerfile:19]: discussed this will be addressed in May release task

File: ReleaseNotes.md [hunk 9]
```diff
@@ -8,6 +8,30 @@ information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeo
...
+  - Add support to specify azure autonomous resource endpoint parameter for the Extension chart for custom metrics
+  - Add autonomous fqdn to endpoint for metrics 
+  - Add devskim as github action and codeql as enabled in ADO pipeline
+  - AgentLogCollection.sh update for collecting more detailed logs
+  - Fix pod ready condition issue for pods that are job ready 
+  - Updating packages for go vulnerabilities
+  - Remove the necessary askcoin references in troubleshooter and readme
+  - Change md5 to sha256 for the omslogger
+  - Log message and yaml to sync with AKS RP
+  - Add nil check in kubelet
```
Comment (ganga1980) [inline ReleaseNotes.md:30]: Add below line as well
 - Fix fluent-bit output plugin crash when the tags are missing for the telegraf metrics

File: .trivyignore [hunk 10]
```diff
@@ -18,3 +18,6 @@ CVE-2022-3821
 # ruby vulnerabilities
 CVE-2023-28756
 CVE-2023-28755
+
+# runc vulnerabilities
```
Comment (ganga1980) [inline .trivyignore:22]: this is telegraf vulnerability and move this under telegraf so we can track and handle these through telegraf upgrade,

File: ReleaseNotes.md (hunk 9 above)
Comment (wanlonghenry) [reply ReleaseNotes.md:30]: added

**Review by ganga1980 — APPROVED** [review]

**Pull request merged**